integrations/BingAds/browser.js

```diff
@@ -0,0 +1,70 @@
+import logger from "../../utils/logUtil";
+
+class BingAds {
+  constructor(config) {
+    this.apikey = config.apikey;
+  }
+
+  loadBingadsScript = () => {
+    ((w, d, t, r, u) => {
+      let f, n, i;
+      w[u] = w[u] || [], f = () => {
+        let o = {
+          ti: this.apikey
+        };
+        o.q = w[u], w[u] = new UET(o);
+      }, n = d.createElement(t), n.src = r, n.async = 1, n.onload = n.onreadystatechange = function() {
+        let s = this.readyState;
+        s && s !== "loaded" && s !== "complete" || (f(), n.onload = n.onreadystatechange = null)
+      }, i = d.getElementsByTagName(t)[0], i.parentNode.insertBefore(n, i)
+    })(window, document, "script", "https://bat.bing.com/bat.js", "uetq");
+  }
+
+  init = () => {
+    this.loadBingadsScript();
+    logger.debug("===in init BingAds===");
+  }
+
+  isLoaded = () => {
+    logger.debug("in BingAds isLoaded");
+    return (!!window.uetq && window.uetq.push !== Array.prototype.push);
+  }
+
+  isReady = () => {
+    logger.debug("in BingAds isReady");
+    return !!(window.uetq && window.uetq.push !== Array.prototype.push);
+  }
+
+  /*
+    Visit here(for details Parameter details): https://help.ads.microsoft.com/#apex/3/en/53056/2
+    Under: What data does UET collect once I install it on my website?
+    It conatins info about parameters ea,ec,gc,gv,el
+  */
+
+  track = (rudderElement) => {
+    const { type, properties } = rudderElement.message;
+    const { category, currency, value, label } = properties;
+    let event = {
+      ea: type
+    };
+    if (category) {
+      event.ec = category;
+    }
+    if (currency) {
+      event.gc = currency;
+    }
+    if (value) {
```

Comment:
this seems like revenue value.
Do we support other keys where revenue can come from?

Reply:
Yes, this is revenue value, it is used to set BingAds **gv** (Variable revenue currency) parameter.
For now, we support only one key.
More detail: Revenue comes from properties fields: e.g.,
`rudderanalytics.track('track conversion', {
                category: 'MyCategory',
                value: 125,
                currency: 'INR',
                order_id: 'order_1'
            });`
What changes will I need to do, please suggest.

Reply:
I think revenue and total can also be considered as valid keys.
@ruchiramoitra plz confirm

Reply:
yes we should consider them too... value, revenue, total